working/augmentation-libraries/parts_with_imports.md

```diff
@@ -0,0 +1,367 @@
+having useful language support when editing the part file.
+
+The (URI) string version is the most useful for making analysis of a part file
+possible and unique, because it uniquely defines the library by the URI that
+the language itself uses to identify files and libraries. It’s technically
+possible to have two separate libraries with the same declared library name,
+which both include the same part file. The language has a restriction against
+having two libraries with the same declared name in the same program *mainly*
+to avoid this particular issue, but that still makes offline analysis of the
+part file a problem.
+
+Pre-feature part files inherit the entire import scope from the library file.
+Each declaration of the library file and each part file is included in the
+library’s declaration scope. It’s viable to think of part files as being
+textually included in the library file. There is a is even a rule against
+declaring a `part` inclusion of the same file more than once, which matches
+perfectly with that way of thinking.
+
+## Feature
+
+This feature allows a part file to have `import`, `export` and `part`
+directives of its own, where `import` directives only affect the part file
+itself, and its transitive part files. A library is defined by the source code
+of its library file and *all* transitively included part files, which can be an
+arbitrarily deep *tree*. A part file inherits the imports and import prefixes
+of its parent file (the library or part file that included it) into its
+top-level scope, but can choose to ignore or shadow those using its own
+imports.
+
+The design goals and principles are:
+
+*   *Backwards compatible*: If a part file has no `import`, `export` or `part`
+    directive, it works just like it always has.
+    *   Because of that, it’s always safe to move one or more declarations into
+        a new part file. _(Although augmentation declarations modifies that
+        slightly.)_
+    *   Similarly it’s always possible and safe to combine a part file with no
+        `import`s back into its parent file.
+*   *Library member declarations are library-global*: All top-level
+    declarations in the library file and all transitive part files are equal,
+    and are all in scope in every file. They introduce declarations into the
+    library’s declaration scope, which is the most significant scope in all
+    files of the library. If there is any conflict, top-level declarations win!
+*   *The unit of ownership is the library*. It’s quite possible for one part
+    file to introduce a conflict with another part file. It always was, but
+    there are new ways too. If that happens, the library owner, who most likely
+    introduced the problem, is expected to fix it. There is no attempt to hide
+    name conflicts between declarations in separate tree-branches of the
+    library structure.
+*   *Import inheritance is a only suggestion*: Aka. other files’ imports cannot
```

Comment:
By "ignore" I just mean not refer to it. 
If the part file needs to re-use the name for something, it can. (But it can't use the anme as an unqualified identifier and hope it resolves to `this.id`, so can't *completely* ignore it.)

So basically: A part file can always *reuse* an inherited name without any issue, so it can ignore *whether* a name is inherited when it chooses to use that name itself.